gax/src/main/java/com/google/api/gax/batching/BatcherImpl.java

```diff
@@ -248,8 +256,10 @@ void add(ElementT element, SettableApiFuture<ElementResultT> result) {
       byteCounter += descriptor.countBytes(element);
     }
 
+    // TODO: Update exception in splitResponse(), So that it does not marks complete batch failed.
     void onBatchSuccess(ResponseT response) {
       try {
+        batcherStats.recordBatchElementsCompletion(results);
```

Comment:
This has to come after splitResponse otherwise the results will never be resolved